File: devops-scm-api/src/main/java/com/tencent/devops/scm/api/exception/ScmApiException.java
package com.tencent.devops.scm.api.exception;

import lombok.Getter;

/**
 * scm api异常基类
 */
@Getter
public class ScmApiException extends RuntimeException {
    Integer statusCode;

    public ScmApiException(String message, int code) {
        super(message);
        this.statusCode = code;
    }

    public ScmApiException(String message) {
        super(message);
    }

    public ScmApiException(String message, Throwable cause) {
        super(message, cause);
    }

    public ScmApiException(Throwable cause) {
        super(cause);
    }
}

File: devops-scm-api/src/main/kotlin/com/tencent/devops/scm/api/exception/ScmApiException.kt
package com.tencent.devops.scm.api.exception

open class ScmApiException : RuntimeException {
    var statusCode: Int? = null

    constructor(message: String?, code: Int) : super(message) {
        this.statusCode = code
    }

    constructor(message: String?) : super(message)

    constructor(message: String?, cause: Throwable?) : super(message, cause)

    constructor(cause: Throwable?) : super(cause)
}
